### PR TITLE
update transfer money

### PR DESCRIPTION
**Pull request by JaskRendix**
PR corrects and simplifies the action transfer money.

When I created it I made a mistake of adding the slug parameter. I recognized the error because I was using it for some maps and I found it very tricky, now it's more simple.

### CODE REVIEW

**Comment by bitcraft** [comment]
im not familiar with the destination for where the funds are going.  have you found that we don't need shop keepers to have a wallet/money?

**Comment by JaskRendix** [comment]
Great point, you're right about the wallet for shops, museums, etc. I have reintroduced the npc_slug
I wanted to simplify the action, before it was a bit confusing, now it's way more simple. 

**before**
`transfer_money -,100,spyder_leather_museum`
here it was already included inside the code that the player was going to lose 100 and the museum was getting 100
**now**
```
transfer_money spyder_leather_museum,+,100
transfer_money player,-,100
```
before it was too much hardcoded, modders need to have the choice if remove or add the money to the shop, museum or npcs, like if there is a gift, etc as well as the player

**Comment by bitcraft** [comment]
im not really sure that the original was complicated at all.  did you ask other modders before making this change?  i suppose it would have been more easily "simplified" by just making the argument for a destination wallet optional.

**Comment by JaskRendix** [comment]
Simplified the original, so the structure doesn't change. The npc is optional.

File: tuxemon/event/actions/transfer_money.py [hunk 1]
```diff
@@ -39,58 +39,51 @@ class MoneyMathAction(EventAction):
...
-                value = 0
-                player.money[destination] = int(value) + amount
-            else:
-                player.money[destination] = int(wallet_slug) + amount
+            player.money["player"] = wallet_player - amount
+            if self.slug is not None:
+                wallet_npc = player.money.get(self.slug)
+                if wallet_npc is None:
+                    player.money[self.slug] = 0 + amount
+                    print(player.money[self.slug])
```
Comment (bitcraft) [inline tuxemon/event/actions/transfer_money.py]: please do not include print statements

Comment (JaskRendix) [reply tuxemon/event/actions/transfer_money.py]: I forgot to remove, it happens

File: tuxemon/event/actions/transfer_money.py [hunk 2]
```diff
@@ -39,58 +39,50 @@ class MoneyMathAction(EventAction):
...
-            if wallet_slug is None:
-                value = 0
-                player.money[destination] = int(value) - amount
-            else:
-                player.money[destination] = int(wallet_slug) - amount
+            player.money["player"] = wallet_player + amount
+            if self.slug is not None:
+                wallet_npc = player.money.get(self.slug)
+                if wallet_npc is None:
+                    player.money[self.slug] = 0 + amount
```
Comment (bitcraft) [inline tuxemon/event/actions/transfer_money.py]: `0 + amount`?  should this be `0 - amount`?

Comment (bitcraft) [reply tuxemon/event/actions/transfer_money.py]: could also be `amount * -1`

File: tuxemon/event/actions/transfer_money.py [hunk 3]
```diff
@@ -39,58 +39,50 @@ class MoneyMathAction(EventAction):
...
-            if wallet_slug is None:
-                value = 0
-                player.money[destination] = int(value) + amount
-            else:
-                player.money[destination] = int(wallet_slug) + amount
+            player.money["player"] = wallet_player - amount
+            if self.slug is not None:
+                wallet_npc = player.money.get(self.slug)
+                if wallet_npc is None:
+                    player.money[self.slug] = 0 + amount
```
Comment (bitcraft) [inline tuxemon/event/actions/transfer_money.py]: why `0 + amount`?

File: tuxemon/event/actions/transfer_money.py (hunk 2 above)
Comment (JaskRendix) [reply tuxemon/event/actions/transfer_money.py]: true, it should be `0 - amount`, I'm going to put `amount * -1`

File: tuxemon/event/actions/transfer_money.py (hunk 3 above)
Comment (JaskRendix) [reply tuxemon/event/actions/transfer_money.py]: true, as above, I'm going to put `amount * 1`

Comment (bitcraft) [reply tuxemon/event/actions/transfer_money.py]: why `amount * 1`?  you could just do `player.money[self.slug] = amount`?

**Review by kerizane — APPROVED** [review]

**Pull request merged**